enterprise/backend/src/metabase_enterprise/advanced_config/file/databases.clj

```diff
@@ -37,18 +37,27 @@
 
 (defn- init-from-config-file!
   [database]
-  ;; assert that we are able to connect to this Database. Otherwise, throw an Exception.
-  (driver.u/can-connect-with-details? (keyword (:engine database)) (:details database) :throw-exceptions)
-  (if-let [existing-database-id (t2/select-one-pk Database :engine (:engine database), :name (:name database))]
+  (if (contains? database :delete)
+    (let [magic-request (format "DELETE_WITH_DEPENDENTS:%s" (:name database))]
+      (when (not= magic-request (:delete database))
+        (throw (ex-info (format "To delete database %s set `delete` to %s" (pr-str (:name database)) (pr-str magic-request))
+                        {:database-name (:name database)})))
+      (when-let [existing-database-id (t2/select-one-pk Database :engine (:engine database), :name (:name database))]
+        (log/info (u/format-color :blue "Deleting Database %s %s" (:engine database) (pr-str (:name database))))
```

Comment:
I think we should include a `log/info` log message for anyone who wishes to use this feature, telling them that deleting database with the config file is subject to breaking changes and they should not rely on its stability.

We usually care quite a lot about not making breaking changes to the config file format, but this is a special case where we are creating the feature specifically for internal use, and I think we should reserve the right to make breaking changes to it.

Reply:
https://github.com/metabase/metabase/pull/46437/commits/839178fc4aa3ac55a0da8d4f5cc01a7d6159b118